pkg/sql/testdata/postgres_test_expected_diffs_on_pg_catalog.json

```diff
@@ -150,20 +150,7 @@
     "pg_statistic": {},
     "pg_statistic_ext_data": {},
     "pg_stats": {},
-    "pg_stats_ext": {},
-    "pg_user": {
-      "valuntil": {
```

Comment:
how did this test pass before? we already were using `timestamptz` for valuntil before this change. why is the json only changing now?

Reply:
pg_user was using timestamp until we fixed it when forEachRoles refactoring. The expected diff was that pg_user.valid until in cockroach was timestamp but in postgress was timestamptz, As this file was already stablishing that difference was expected this didn't fail.

When we fixed this file wasn't updated BUT as there was no difference, then it didn't fail.

Reply:
the test should fail if there is an `expected diff` in the json, but the real implementation does not have the diff. is this not how it works?